### PR TITLE
[enhancement](file-cache) limit the file cache handle num and init the file cache concurrently

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

1. the real value of BE config `file_cache_max_file_reader_cache_size` will be the 1/3 of process's max open file number.
2. use thread pool to create or init the file cache concurrently.
    To solve the issue that when there are lots of files in file cache dir, the starting time of BE will be very slow because
    it will traverse all file cache dirs sequentially.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/io/cache/block/block_file_cache.cpp [hunk 1]
```diff
@@ -205,5 +207,21 @@ size_t IFileCache::file_reader_cache_size() {
     return s_file_name_to_reader.size();
 }
 
+void IFileCache::init() {
+    struct rlimit limit;
+    if (getrlimit(RLIMIT_NOFILE, &limit) != 0) {
```
[the remark this answers is not in the document]
Comment (morningman) [reply be/src/io/cache/block/block_file_cache.cpp:212]: if fail, the `LOG(FATAL)` will crash the BE.

**Comment by morningman** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.67 seconds
 stream load tsv:          515 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162288766 Bytes

**Comment by morningman** [comment]
run external

**Comment by zhangguoqiang666** [comment]
run external

**Comment by zhangguoqiang666** [comment]
run buildall

**Comment by zhangguoqiang666** [comment]
run external

**Comment by zhangguoqiang666** [comment]
run external

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.83 seconds
 stream load tsv:          517 seconds loaded 74807831229 Bytes, about 137 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162138449 Bytes

**Comment by morningman** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.37 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162144885 Bytes

**Review by AshinGau — APPROVED** [review]

**Comment by AshinGau** [comment]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**